lib/widgets/theme.dart

```diff
@@ -554,10 +554,19 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
   }
 }
 
+// This is taken from:
+//   https://github.com/zulip/zulip/blob/b248e2d93/web/src/stream_data.ts#L40
+const kDefaultChannelColorSwatchBaseColor = 0xffc2c2c2;
+
 /// The theme-appropriate [ChannelColorSwatch] based on [subscription.color].
 ///
+/// If [subscription] is null, [ChannelColorSwatch] will be based on
+/// [kDefaultChannelColorSwatchBaseColor].
+///
 /// For how this value is cached, see [ChannelColorSwatches.forBaseColor].
-ChannelColorSwatch colorSwatchFor(BuildContext context, Subscription subscription) {
+// TODO(#188) pick different colors for unsubscribed channels
```

Comment:
Looking into the design requirement a bit more, we initially fall back
    to this color because we didn't have the logic to generate channel
    colors:

As discussed at https://github.com/zulip/zulip-flutter/pull/1491#issuecomment-2845893203 , the thing we didn't have then is exactly the thing we don't have now either, the one this TODO points out :slightly_smiling_face:.